pkg/batch/batch.go

```diff
@@ -38,6 +39,9 @@ func newEntry(ctx context.Context, value interface{}) *entry {
 
 // setResult sets the result of a batch entry.
 func (e *entry) setResult(value interface{}, err error) {
+	if attachDiskParams, ok := e.value.(provider.AttachDiskParams); ok {
```

Comment:
Rather than creating an upward dependency on the `provider` package through `AttachDiskParms`, you should instead define an interface, like [`Closer`](https://pkg.go.dev/io#Closer) at the `batch` package level that a type can implement. You can then type assert to check for the interface and call the `Close` method. (Having the interface share the same method declarations as `Closer` means that *any* data type implementing the `Close` method could satisfy the interface requirements and be used to clean up the value.)